pkg/apis/infra/v1alpha1/cluster.go

```diff
@@ -51,6 +51,13 @@ type CustomClusterSpec struct {
 
 	// CNIConfig is the configuration for the CNI of the cluster on VMs.
 	CNI CNIConfig `json:"cni"`
+
+	// KubesprayVersion is the version of Kubespray used by the customCluster manager worker pod.
+	KubesprayVersion string `json:"kubesprayVersion"`
```

Comment:
I can not see the need to expose to enduser

Reply:
ok, we can change it to default v2.20.0 which support k8s version from v1.22.0 to v1.24.6, since the latest v2.21.0 (support k8s v1.23.0 to 1.25.6 )is failed. https://github.com/kubernetes-sigs/kubespray/issues/9833#issuecomment-1475517465